api/src/security/auth-utils.test.ts

```diff
@@ -270,7 +278,161 @@ describe('authorize', function () {
+      expect(actualError).instanceOf(HTTP403);
+    }
+  });
+
+  it('authorizes a user with valid roles', async function () {
+    sinon.stub(auth_utils, 'getSystemUser').resolves({
+      id: 0,
+      user_identifier: 'somebody',
+      role_ids: [],
+      role_names: []
+    });
+
+    try {
+      const result = await auth_utils.authorize({ keycloak_token: 'any token' }, ['abc']);
+      expect(result).to.be.true;
+    } catch {
+      expect.fail();
     }
   });
 });
+
+describe('authenticate', function () {
+  it('throws HTTP401 when authorization headers were null or missing', async function () {
+    try {
+      await auth_utils.authenticate(undefined);
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP401);
+    }
+
+    try {
+      await auth_utils.authenticate({});
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP401);
+    }
+
+    try {
+      await auth_utils.authenticate({
+        headers: {}
+      });
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP401);
+    }
+  });
+
+  it('throws HTTP401 when authorization header containing valid bearer token', async function () {
+    try {
+      await auth_utils.authenticate({
+        headers: {
+          authorization: 'Not a bearer token'
+        }
+      });
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP401);
+    }
+
+    try {
+      await auth_utils.authenticate({
+        headers: {
+          authorization: 'Bearer '
+        }
+      });
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP401);
+    }
+
+    try {
+      await auth_utils.authenticate({
+        headers: {
+          authorization: 'Bearer not-encoded'
+        }
+      });
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP401);
+    }
+
+    try {
+      await auth_utils.authenticate({
+        headers: {
+          // sample encoded json web token from jwt.io (without kid header)
+          authorization:
+            'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c'
+        }
+      });
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP401);
+    }
+
+    try {
+      await auth_utils.authenticate({
+        headers: {
+          // sample encoded json web token from jwt.io (with kid header)
+          authorization:
+            'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6InNvbWUgaWQifQ.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.rXYCCTRmi7EAHVRy1QqHvysMxLZaH4EsI226hrJLtgM'
+        }
+      });
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP401);
+    }
+  });
+
+  //TODO:  figure out how to stub the callback parameter of the function
+
+  // import * as jsonwebtoken from 'jsonwebtoken';
+  // it('returns a validated token through verifyToken', async function () {
```

Comment:
If you just want to test the return value of this when its called by the authenticate function, then you can just stub out the entire `verifyToken` function, and make it return w/e you want. 

And then we can test the verifyToken function directly, in its own set of unit tests.